taskpools/primitives/barriers_posix.nim

```diff
@@ -18,8 +18,12 @@ when defined(osx):
   export PthreadAttr, PthreadBarrier, Errno, PTHREAD_BARRIER_SERIAL_THREAD
 else:
   type
-    PthreadAttr* {.byref, importc: "pthread_attr_t", header: "<sys/types.h>".} = object
-    PthreadBarrier* {.byref, importc: "pthread_barrier_t", header: "<sys/types.h>".} = object
+    PthreadAttr* {.importc: "pthread_attr_t", header: "<sys/types.h>".} = object
```

Comment:
The `byref` is important, it doesn't make sense to copy a barrier parameter

Reply:
`byref` is a bit of a hack really - ie it creates an `object` (which is a value-based type) and then _sometimes_ wires it to be passed by pointer - `c2nim` handles this more consistently by using `var` / `ptr` explicitly in functions instead  of hiding it in the type - without this, you need to lie in the function signature (by making it look like you're passing in a value, when in fact, it's treated as a pointer).

over time, ideally, we'd use the style that `c2nim` promotes  (which does not use byref - instead, it explicitly states the parameter passing style in each function) - this would allow "more" automatic generation of these ABI wrappers.

Reply:
well, `c2nim` goes in the other direction: it annotates everything with `bycopy` (https://github.com/status-im/nim-bearssl/blob/master/bearssl/abi/bearssl_aead.nim#L15) and then introduces the pointer on the function definition..

this line of thinking is similar to `T = ref object` vs `param: ref T` - the latter is explicit at the function site instead of the object definition, and for most code, we've chosen to not use `ref object` because nim "naturally" does not work by ref.